src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -489,6 +489,25 @@ def _get_extension_suppressions(mod_loaders):
 
         return self.command_table
 
+    def sort_command_loaders(self, command_loaders):
+        module_command_loaders = []
+        extension_command_loaders = []
+
+        # Separate module and extension command loaders
+        for loader in command_loaders:
+            if loader.__module__.startswith('azext'):
+                extension_command_loaders.append(loader)
+            else:
+                module_command_loaders.append(loader)
+        
+        # Sort name in each command loader list
+        module_command_loaders.sort(key=lambda loader: loader.__class__.__name__)
+        extension_command_loaders.sort(key=lambda loader: loader.__class__.__name__)
+
+        # Module first, then extension
+        sorted_command_loaders = module_command_loaders + extension_command_loaders
```

Comment:
The loading order of modules and extensions matters - modules must be loaded first. 

Otherwise, an extension can maliciously or accidentally affect other modules. For example, an extension can declare 

```py
def load_arguments(self, _):
    with self.argument_context('resource') as c:
        c.ignore('_subscription')  # hide global subscription param
```

thus corrupting the global `--subscription` for all `az resource`'s commands.